trin-beacon/src/jsonrpc.rs

```diff
@@ -25,14 +25,13 @@ use crate::network::BeaconNetwork;
 /// Handles Beacon network JSON-RPC requests
 pub struct BeaconRequestHandler {
     pub network: Arc<RwLock<BeaconNetwork>>,
```

Comment:
Since you are at this topic, is there a reason why network should be wrapped around RwLock?
If I'm not mistaken, every possible usage would need `read` access anyway and there is no functionality that needs `write` access.

Similar in few other places.

Reply:
I didn't look into it yet, but you are right it would probably be good to look into this

Reply:
If we can remove it I will handle it in a follow up PR as this PR has already been reviewed/approved with the current changes